### PR TITLE
Improve JSON Unmarshalling for Duration Type in fs Package

### LINKED ISSUE #3783: Could not remove MinAge filter when using 'rclone rc'
#### Problem statement:
I am trying to set and reset the MinAge through the options/set command as follow:
```rclone rc options/set --json '{"filter":{"MinAge":<age in nano sec>}}' --rc-addr=<ip:port> --rc-user=<user> --rc-pass=<pass>```
No problem to set the MinAge. But could not remove such filter and ensure there is no MinAge limit. I mean when you run rclone rcd there is no filter regarding the min age. And the MinAge is 9223372036854776000 by default.

#### rclone version
rclone v1.49.1
- os/arch: windows/amd64
- go version: go1.12.3

#### OS
Multiple OS and versions

####  Storage
Swift

#### The command  trying to run
`rclone rc options/set --json '{"filter":{"MinAge": 9223372036854775807}}"`

#### log

ERROR : rc: "options/set": error: failed to write options from block "filter": Reshape failed to Unmarshal: json: cannot unmarshal number 9223372036854776000 into Go struct field Opt.MinAge of type fs.Duration

### PR DESCRIPTION
**Pull request by KDreynolds**
Enhanced the UnmarshalJSON method for the Duration type to correctly handle the special string 'off' and ensure large integers are parsed accurately without floating-point rounding errors. This resolves issues with setting and removing the MinAge filter through the rclone rc command.

Fixes #3783

Thank you very much for contributing code or documentation to rclone! Please
fill out the following questions to make it easier for us to review your
changes.

You do not need to check all the boxes below all at once, feel free to take
your time and add more commits. If you're done and ready for review, please
check the last box.

#### What is the purpose of this change?

This Pull Request introduces enhancements to the `Duration` type's `UnmarshalJSON` method within the `fs` package to correctly handle the special string "off" and ensure large integers are parsed accurately without floating-point rounding errors. This change is crucial for allowing users to set and remove the MinAge filter through the `rclone rc` command without encountering errors.

### Problem
When attempting to remove the MinAge filter by setting its value to the maximum integer, users encountered JSON unmarshalling errors due to the way large integers were handled.

### Solution
By checking for the special "off" string and parsing large integers directly as `int64`, we avoid the rounding errors associated with floating-point conversions and ensure compatibility with the expected `Duration` values.

### Impact
This fix allows for more flexible and error-free configuration of filters through the `rclone rc` command, enhancing the user experience.

Fixes #3783

#### Was the change discussed in an issue or in the forum before?

issue #3783 

#### Checklist

- [x] I have read the [contribution guidelines](https://github.com/rclone/rclone/blob/master/CONTRIBUTING.md#submitting-a-new-feature-or-bug-fix).
- [x] I have added tests for all changes in this PR if appropriate.
- [x] I have added documentation for the changes if appropriate.
- [x] All commit messages are in [house style](https://github.com/rclone/rclone/blob/master/CONTRIBUTING.md#commit-messages).
- [x] I'm done, this Pull Request is ready for review :-)

### CODE REVIEW

**Comment by KDreynolds** [comment]
Ok @ncw, I think this works for a solution, please let me know if this needs any more work. Thanks!

File: fs/parseduration.go [hunk 1]
```diff
@@ -231,10 +232,30 @@ func (d Duration) Type() string {
 
 // UnmarshalJSON makes sure the value can be parsed as a string or integer in JSON
 func (d *Duration) UnmarshalJSON(in []byte) error {
-	return UnmarshalJSONFlag(in, d, func(i int64) error {
-		*d = Duration(i)
+	// Check if the input is a string value.
+	if in[0] == '"' {
```
Comment (ncw) [inline fs/parseduration.go:236]: You should check `len(in) > 0` before checking `in[0]`.  In fact for this if branch you want `len(in) >= 2` I think.

So I think this should say

```go
if len(in) >= 2 && in[0] == '"' && in[len(in)-1] == '"' {
```

File: fs/parseduration.go [hunk 2]
```diff
@@ -231,10 +232,30 @@ func (d Duration) Type() string {
...
+		strVal := string(in[1 : len(in)-1]) // Remove the quotes
+		// Handle the special "off" case.
+		if strVal == "off" {
```
Comment (ncw) [inline fs/parseduration.go:239]: I'm pretty sure ParseDuration parses `off` so this shouldn't be needed

**Review by ncw — COMMENTED** [review]
I think this is looking good - thank you :-)

I put some minor nits inline.

**Comment by KDreynolds** [comment]
ok i think I made the proper updates, lmk if not. Thanks again!

**Review by ncw — APPROVED** [review]
This looks great now - thank you.

**Pull request merged**